mysqltest/config.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysqltest
+
+import "log"
+
+// This file implements the "functional options" pattern.
+
+type config struct {
+	killAfterSec   int // This is in integer seconds because that's what Docker takes.
+	mySQLVersion   string
+	progressLogger Logger
+}
+
+type Logger interface {
+	Printf(fmtStr string, args ...any)
+}
+
+func makeDefaultConfig() *config {
+	return &config{
+		killAfterSec:   10 * 60,
+		mySQLVersion:   "5.7",
+		progressLogger: stdlibLogger{},
```

Comment:
Any reason not to use a pointer?

```suggestion
		progressLogger: &stdlibLogger{},
```

Reply:
I don't have a preference either way. Changed to a pointer.